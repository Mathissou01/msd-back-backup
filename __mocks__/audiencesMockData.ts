import { GetAudiencesByContractIdDocument } from "../src/graphql/codegen/generated-types";

export const getNoAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [],
        },
      },
    },
  },
];

export const getSingleAudienceMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getTwoAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];

export const getAllAudiencesMockedData = [
  {
    request: {
      query: GetAudiencesByContractIdDocument,
      variables: {
        filters: {
          contract: {
            id: {
              eq: undefined,
            },
          },
          isActive: {
            eq: true,
          },
        },
      },
    },
    result: {
      data: {
        audiences: {
          data: [
            {
              id: 1,
              attributes: {
                isActive: true,
                type: "Particuliers",
                __typename: "Audience",
              },
            },
            {
              id: 2,
              attributes: {
                isActive: true,
                type: "Collectifs",
                __typename: "Audience",
              },
            },
            {
              id: 3,
              attributes: {
                isActive: true,
                type: "Professionnels",
                __typename: "Audience",
              },
            },
          ],
        },
      },
    },
  },
];
