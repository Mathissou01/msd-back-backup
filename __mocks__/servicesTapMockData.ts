import { GetServicesBlockTabDocument } from "../src/graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetServicesBlockTabDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              id: "1",
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      servicesBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            titleContent: "Les services de ma ville",
                            serviceLinks: [
                              {
                                name: "Carte",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                              {
                                name: "Collecte à mon adresse",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                              {
                                name: "Guide du tri",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                              {
                                name: "Faire une demande",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                              {
                                name: "Réduire mes déchets",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                              {
                                name: "Valoriser mes déchets",
                                isDisplayed: true,
                                picto: {
                                  data: null,
                                },
                              },
                            ],
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
      },
    },
  },
];
