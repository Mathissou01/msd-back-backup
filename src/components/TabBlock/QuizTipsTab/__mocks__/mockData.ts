import { GetQuizAndTipsBlockDocument } from "../../../../graphql/codegen/generated-types";

export const defaultMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    result: {
      data: {
        contractCustomizations: {
          data: [
            {
              attributes: {
                homepage: {
                  data: {
                    attributes: {
                      quizAndTipsBlock: {
                        data: {
                          id: "1",
                          attributes: {
                            title: "Titre modifi√©",
                            displayBlock: false,
                            displayQuiz: false,
                            quiz: {
                              data: {
                                id: "2",
                                attributes: {
                                  title: "Nom du quiz",
                                  publishedAt: "2022-11-20T02:34:14.535Z",
                                },
                              },
                            },
                            displayTips: false,
                            tips: {
                              data: [],
                            },
                          },
                        },
                      },
                    },
                  },
                },
              },
            },
          ],
        },
        services: {
          data: [
            {
              attributes: {
                serviceInstance: [
                  {
                    __typename: "ComponentMsdEditorial",
                    id: "4",
                    editorialServices: {
                      data: [
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoQuizzesSubService",
                                isActivated: true,
                                quizzes: {
                                  data: [
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Nom du quiz",
                                        publishedAt: "2022-11-20T02:34:14.535Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "truc",
                                        publishedAt: "2022-11-18T10:16:08.936Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                        {
                          attributes: {
                            subServiceInstance: [
                              {
                                __typename: "ComponentEditoTipsSubService",
                                isActivated: true,
                                tips: {
                                  data: [
                                    {
                                      id: "5",
                                      attributes: {
                                        title: "nah",
                                        publishedAt: "2022-11-19T00:14:37.403Z",
                                      },
                                    },
                                    {
                                      id: "2",
                                      attributes: {
                                        title: "Titre de l'astuce 1",
                                        publishedAt: "2022-11-19T00:14:08.232Z",
                                      },
                                    },
                                    {
                                      id: "1",
                                      attributes: {
                                        title: "Titre de l'astuce 2",
                                        publishedAt: "2022-11-18T10:17:11.158Z",
                                      },
                                    },
                                    {
                                      id: "3",
                                      attributes: {
                                        title: "Titre de l'astuce 3",
                                        publishedAt: "2022-11-19T00:14:20.490Z",
                                      },
                                    },
                                    {
                                      id: "4",
                                      attributes: {
                                        title: "Titre de l'astuce 4",
                                        publishedAt: "2022-11-19T00:14:30.273Z",
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
          ],
        },
      },
    },
  },
];

export const errorMockData = [
  {
    request: {
      query: GetQuizAndTipsBlockDocument,
      variables: {
        contractId: "1",
      },
    },
    error: new Error("An error occured"),
  },
];
