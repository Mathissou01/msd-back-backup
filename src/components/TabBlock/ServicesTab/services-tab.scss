@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.buttons" as buttons;
@use "src/styles/02-tools/tools.typography" as typo;

.c-ServicesTab {
  position: relative;
  width: 100%;
  height: 100%;

  &__Title {
    @include typo.heading(2);

    margin-bottom: 18px;
    color: colors.$expert-blue;
  }

  &__Form {
    display: flex;
    flex-direction: column;
    gap: 58px;
    width: 100%;
  }

  &__Block {
    display: flex;
    flex-direction: column;
    gap: 18px;
    align-items: flex-start;
    width: 348px;
    padding: 0;
  }

  &__Group {
    width: 100%;
  }

  &__ServiceLinks {
    display: flex;
    flex-direction: column;
    gap: 30px;
  }

  &__AddButton {
    display: flex;
    flex-direction: row;
  }

  &__NotDisplayServices {
    display: flex;
    flex-direction: column;
    gap: 18px;
    padding: 0;
  }

  &__Buttons {
    display: flex;
    gap: 16px;
  }
}
