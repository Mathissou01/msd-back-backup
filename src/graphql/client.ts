import {
  ApolloClient,
  InMemoryCache,
  NormalizedCacheObject,
} from "@apollo/client";

let client: ApolloClient<NormalizedCacheObject>;

if (process.env.NEXT_PUBLIC_MOCK === "true") {
  /* With .env variable NEXT_PUBLIC_MOCK=true, return mock graphql client with fake response data */
  client = (await import("./mockedClient")).default(1000);
} else {
  client = await new ApolloClient({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql`,
    cache: new InMemoryCache({
      typePolicies: {
        UploadFolder: {
          merge: true,
        },
        UploadFile: {
          merge: true,
        },
        Contract: {
          merge: true,
        },
        ContractCustomization: {
          merge: true,
        },
        New: {
          merge: true,
        },
        Tip: {
          merge: true,
        },
        RecyclingGuideService: {
          merge: true,
        },
        WasteForm: {
          merge: true,
        },
        WasteFamily: {
          merge: true,
        },
        Flow: {
          merge: true,
        },
        FlowColor: {
          merge: true,
        },
        DropOffMap: {
          merge: true,
        },
        Request: {
          merge: true,
        },
        RequestAggregate: {
          merge: true,
        },
      },
    }),
  });
}

export default client;
