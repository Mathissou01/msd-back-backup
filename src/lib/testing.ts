import { prettyDOM } from "@testing-library/react";

export function removeUnstableHtmlPropertiesForTesting(
  htmlElement: HTMLElement,
  attributeName: string,
) {
  const domHTML = prettyDOM(htmlElement, Infinity).toString();
  if (!domHTML) return undefined;
  const regex = new RegExp(`${attributeName}(.*)"(.*)"`, "g");
  return domHTML.replaceAll(regex, "");
}
