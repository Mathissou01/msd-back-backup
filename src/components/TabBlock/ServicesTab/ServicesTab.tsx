import { FormProvider, useForm } from "react-hook-form";
import { FieldValues } from "react-hook-form/dist/types/fields";
import React, { useEffect, useState } from "react";
import FormInput from "../../Form/FormInput/FormInput";
import {
  GetServicesBlockTabDocument,
  useGetServicesBlockTabQuery,
  useUpdateServicesBlockTabMutation,
} from "../../../graphql/codegen/generated-types";
import { useContract } from "../../../hooks/useContract";
import CommonButton from "../../Common/CommonButton/CommonButton";
import { IServiceLink } from "../../../lib/service-links";
import FormServiceLinks from "../../Form/FormServiceLinks/FormServiceLinks";
import ServicesTabAddButton from "./ServicesTabAddButton/ServicesTabAddButton";
import "./services-tab.scss";
import CommonLoader from "../../Common/CommonLoader/CommonLoader";
import { useFocusFirstElement } from "../../../hooks/useFocusFirstElement";
import { extractServicesBlock } from "../../../lib/graphql-data";

interface IServicesBlock {
  id: string;
  titleContent: string;
  serviceLinks: Array<IServiceLink>;
}

export default function ServicesTab() {
  /* Static Data */
  const title = "Services";
  const maxLimitDisplay = 6;
  const formLabels = {
    titleContent: "Titre du bloc",
    blockDisplayedLabel:
      "Ordre d’affichage des services sur la page d’accueil (accès rapide)",
    secondaryLabel: `Vous pouvez choisir d’afficher jusqu’à ${maxLimitDisplay} blocs maximum`,
    blockNotDisplayedLabel: "Bloc non affichés",
    blockNotDisplayedSecondaryLabel:
      "Nombre maximum de blocs atteints. Pour afficher un autre bloc, vous devez masquer un bloc ci-dessus.",
    editModalTitle: "Encart service",
    editModalNameLabel: "Texte du bouton",
    submitButtonLabel: "Enregistrer les modifications",
    cancelButtonLabel: "Annuler les modifications",
  };

  /* Method */
  function onModalSubmit(data: FieldValues) {
    const modalData = data["createModal"];
    if (modalData && servicesBlockData?.serviceLinks) {
      const newLink: IServiceLink = {
        type: "ComponentLinksExternal",
        localId: servicesBlockData.serviceLinks.length,
        isDisplayed: false,
        name: modalData["name"],
        externalLink: modalData["externalLink"],
        picto: modalData["picto"],
      };
      const newServiceLinks = [...servicesBlockData.serviceLinks, newLink];
      setServicesBlockData({
        ...servicesBlockData,
        serviceLinks: newServiceLinks,
      });
      form.setValue("serviceLinks", newServiceLinks, { shouldDirty: true });
    }
  }

  async function onSubmitValid(submitData: FieldValues) {
    if (submitData["id"]) {
      const returnTitleContent = submitData["titleContent"];
      const returnValues = submitData["serviceLinks"].map(
        (link: IServiceLink) => {
          return {
            __typename: link.type,
            isDisplayed: link.isDisplayed,
            name: link.name,
            externalLink: link.externalLink,
            picto: link.picto,
          };
        },
      );
      const variables = {
        updateServicesBlockId: submitData["id"],
        data: {
          titleContent: returnTitleContent,
          serviceLinks: returnValues,
        },
      };
      return updateServicesBlockTab({
        variables,
        refetchQueries: [
          {
            query: GetServicesBlockTabDocument,
            variables: { contractId },
          },
          "getServicesBlockTab",
        ],
      });
    }
  }

  function onCancel() {
    form.reset();
  }

  /* External Data */
  const { contractId } = useContract();
  const { loading, error, data } = useGetServicesBlockTabQuery({
    variables: { contractId },
  });
  const [
    updateServicesBlockTab,
    { loading: mutationLoading, error: mutationError },
  ] = useUpdateServicesBlockTabMutation();
  /* Local Data */
  const [servicesBlockData, setServicesBlockData] = useState<IServicesBlock>();
  const formValidationMode = "onChange";
  const form = useForm({
    mode: formValidationMode,
  });
  const { handleSubmit, formState } = form;
  const { isDirty, isSubmitting } = formState;

  useEffect(() => {
    if (data) {
      const serviceBlockMapped = extractServicesBlock(data);
      if (
        serviceBlockMapped.id &&
        serviceBlockMapped.titleContent &&
        serviceBlockMapped.serviceLinks
      ) {
        const mappedData = {
          id: serviceBlockMapped.id,
          titleContent: serviceBlockMapped.titleContent,
          serviceLinks: serviceBlockMapped.serviceLinks,
        };
        setServicesBlockData(mappedData);
        form.reset(mappedData);
      }
    }
  }, [form, data]);

  return (
    <div
      className="c-ServicesTab"
      id="panel-services"
      role="tabpanel"
      aria-labelledby="tab-services"
    >
      <CommonLoader
        isLoading={loading || isSubmitting || mutationLoading}
        isShowingContent={isSubmitting || mutationLoading}
        hasDelay={isSubmitting || mutationLoading}
        errors={[error, mutationError]}
      >
        <h2 className="c-ServicesTab__Title">{title}</h2>
        <FormProvider {...form}>
          <form
            className="c-ServicesTab__Form"
            onSubmit={handleSubmit(onSubmitValid)}
            ref={useFocusFirstElement()}
          >
            <div className="c-ServicesTab__Block">
              <div className="c-ServicesTab__Group">
                <FormInput
                  type="text"
                  name="titleContent"
                  label={formLabels.titleContent}
                  isRequired={true}
                  isDisabled={mutationLoading}
                  defaultValue={servicesBlockData?.titleContent}
                />
              </div>
            </div>
            <div className="c-ServicesTab__ServiceLinks">
              <FormServiceLinks
                name="serviceLinks"
                label={formLabels.blockDisplayedLabel}
                secondaryLabel={formLabels.secondaryLabel}
                editModalTitle={formLabels.editModalTitle}
                editModalNameLabel={formLabels.editModalNameLabel}
                splitLabel={formLabels.blockNotDisplayedLabel}
                splitSecondaryLabel={formLabels.blockNotDisplayedSecondaryLabel}
                maxLimitIsDisplayed={maxLimitDisplay}
                isDisabled={mutationLoading}
                isSplitDisplay={true}
              />
              <ServicesTabAddButton onSubmit={onModalSubmit} />
            </div>
            <div className="c-ServicesTab__Buttons">
              <CommonButton
                type="submit"
                label={formLabels.submitButtonLabel}
                style="primary"
                picto="check"
                isDisabled={!isDirty}
              />
              <CommonButton
                type="button"
                label={formLabels.cancelButtonLabel}
                picto="cross"
                onClick={onCancel}
                isDisabled={!isDirty}
              />
            </div>
          </form>
        </FormProvider>
      </CommonLoader>
    </div>
  );
}
