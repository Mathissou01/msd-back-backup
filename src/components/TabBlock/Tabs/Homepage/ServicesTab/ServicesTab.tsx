import { FormProvider, useForm } from "react-hook-form";
import { FieldValues } from "react-hook-form/dist/types/fields";
import React, { useEffect, useState } from "react";
import FormInput from "../../../../Form/FormInput/FormInput";
import {
  AudienceEntity,
  useGetServicesBlocksByContractIdAndAudienceIdQuery,
  useUpdateServicesBlockByIdMutation,
} from "../../../../../graphql/codegen/generated-types";
import { IServiceLink } from "../../../../../lib/service-links";
import { extractServicesBlock } from "../../../../../lib/graphql-data";
import { getRightsByLabel } from "../../../../../lib/user";
import { useFocusFirstElement } from "../../../../../hooks/useFocusFirstElement";
import { useUser } from "../../../../../hooks/useUser";
import { useContract } from "../../../../../hooks/useContract";
import CommonButton from "../../../../Common/CommonButton/CommonButton";
import CommonLoader from "../../../../Common/CommonLoader/CommonLoader";
import FormServiceLinks from "../../../../Form/FormServiceLinks/FormServiceLinks";
import ServicesTabAddButton from "./ServicesTabAddButton/ServicesTabAddButton";
import "./services-tab.scss";

interface IServicesBlock {
  id: string;
  titleContent: string;
  serviceLinks: Array<IServiceLink>;
}

interface IServicesTabProps {
  audience?: AudienceEntity;
}

export default function ServicesTab({ audience }: IServicesTabProps) {
  /* Static Data */
  const title = "Services";
  const maxLimitDisplay = 6;
  const formLabels = {
    titleContent: "Titre du bloc",
    blockDisplayedLabel:
      "Ordre d’affichage des services sur la page d’accueil (accès rapide)",
    secondaryLabel: `Vous pouvez choisir d’afficher jusqu’à ${maxLimitDisplay} blocs maximum, à choisir en fonction du menu défini préalablement`,
    blockNotDisplayedLabel: "Bloc non affichés",
    blockNotDisplayedSecondaryLabel:
      "Nombre maximum de blocs atteints. Pour afficher un autre bloc, vous devez masquer un bloc ci-dessus.",
    editModalTitle: "Encart service",
    editModalNameLabel: "Texte du bouton",
    submitButtonLabel: "Enregistrer les modifications",
    cancelButtonLabel: "Annuler les modifications",
  };

  /* Method */
  function onModalSubmit(data: FieldValues) {
    if (
      data["createModal_name"] &&
      data["createModal_externalLink"] &&
      data["createModal_picto"] &&
      servicesBlockData?.serviceLinks
    ) {
      const newLink: IServiceLink = {
        type: "ComponentLinksExternal",
        isDisplayed: false,
        name: data["createModal_name"],
        externalLink: data["createModal_externalLink"],
        picto: data["createModal_picto"],
      };
      const newServiceLinks = [...servicesBlockData.serviceLinks, newLink];
      setServicesBlockData({
        ...servicesBlockData,
        serviceLinks: newServiceLinks,
      });
      form.setValue("serviceLinks", newServiceLinks, { shouldDirty: true });
    }
  }

  async function onSubmitValid(submitData: FieldValues) {
    if (submitData["id"]) {
      const returnTitleContent = submitData["titleContent"];
      const returnValues = submitData["serviceLinks"].map(
        (link: IServiceLink) => {
          return {
            __typename: link.type,
            isDisplayed: link.isDisplayed,
            name: link.name,
            externalLink: link.externalLink,
            picto: link.picto?.id,
          };
        },
      );
      const variables = {
        updateServicesBlockId: submitData["id"],
        data: {
          titleContent: returnTitleContent,
          serviceLinks: returnValues,
        },
      };
      return updateServicesBlock({
        variables,
      });
    }
  }

  function onCancel() {
    form.reset();
  }

  /* External Data */
  const { contractId } = useContract();
  const getServicesBlockVariables: {
    contractId: string;
    audienceId: string;
  } = {
    contractId,
    audienceId: "",
  };
  if (audience && audience.id) {
    getServicesBlockVariables.audienceId = audience.id;
  }
  const { loading, error, data } =
    useGetServicesBlocksByContractIdAndAudienceIdQuery({
      variables: getServicesBlockVariables,
      fetchPolicy: "network-only",
    });
  const [
    updateServicesBlock,
    { loading: mutationLoading, error: mutationError },
  ] = useUpdateServicesBlockByIdMutation({
    refetchQueries: ["getServicesBlocksByContractIdAndAudienceId"],
    awaitRefetchQueries: true,
  });

  /* Local Data */
  const { userRights } = useUser();
  const userPermissions = getRightsByLabel("Homepage", userRights);
  const [servicesBlockData, setServicesBlockData] = useState<IServicesBlock>();
  const formValidationMode = "onChange";
  const form = useForm({
    mode: formValidationMode,
  });
  const { handleSubmit, formState } = form;
  const { isDirty, isSubmitting } = formState;

  useEffect(() => {
    if (data) {
      const serviceBlockMapped = extractServicesBlock(data);
      if (
        serviceBlockMapped.id &&
        serviceBlockMapped.titleContent &&
        serviceBlockMapped.serviceLinks
      ) {
        const mappedData = {
          id: serviceBlockMapped.id,
          titleContent: serviceBlockMapped.titleContent,
          serviceLinks: serviceBlockMapped.serviceLinks.map(
            (serviceLink, index) => {
              serviceLink.localId = `local-${index}`;
              return serviceLink;
            },
          ),
        };
        setServicesBlockData(mappedData);
        form.reset(mappedData);
      }
    }
  }, [form, data]);

  return (
    <div
      className="c-ServicesTab"
      id="panel-services"
      role="tabpanel"
      aria-labelledby="tab-services"
    >
      <CommonLoader
        isLoading={loading || isSubmitting || mutationLoading}
        isShowingContent={isSubmitting || mutationLoading}
        hasDelay={isSubmitting || mutationLoading}
        errors={[error, mutationError]}
      >
        <h2 className="c-ServicesTab__Title">{title}</h2>
        <FormProvider {...form}>
          <form
            className="c-ServicesTab__Form"
            onSubmit={handleSubmit(onSubmitValid)}
            ref={useFocusFirstElement()}
          >
            <div className="c-ServicesTab__Block">
              <div className="c-ServicesTab__Group">
                <FormInput
                  type="text"
                  name="titleContent"
                  label={formLabels.titleContent}
                  isRequired={true}
                  isDisabled={mutationLoading || !userPermissions.update}
                  defaultValue={servicesBlockData?.titleContent}
                />
              </div>
            </div>
            <div className="c-ServicesTab__ServiceLinks">
              <FormServiceLinks
                name="serviceLinks"
                label={formLabels.blockDisplayedLabel}
                secondaryLabel={formLabels.secondaryLabel}
                editModalTitle={formLabels.editModalTitle}
                editModalNameLabel={formLabels.editModalNameLabel}
                isDisabled={mutationLoading || !userPermissions.update}
                isSortByIsDisplayed={true}
                isSplitDisplay={true}
                splitLabel={formLabels.blockNotDisplayedLabel}
                splitSecondaryLabel={formLabels.blockNotDisplayedSecondaryLabel}
                maxLimitIsDisplayed={maxLimitDisplay}
              />
              <ServicesTabAddButton onSubmit={onModalSubmit} />
            </div>
            <div className="c-ServicesTab__Buttons">
              <CommonButton
                type="submit"
                label={formLabels.submitButtonLabel}
                style="primary"
                picto="check"
                isDisabled={!isDirty || !userPermissions.update}
              />
              <CommonButton
                type="button"
                label={formLabels.cancelButtonLabel}
                picto="cross"
                onClick={onCancel}
                isDisabled={!isDirty}
              />
            </div>
          </form>
        </FormProvider>
      </CommonLoader>
    </div>
  );
}
