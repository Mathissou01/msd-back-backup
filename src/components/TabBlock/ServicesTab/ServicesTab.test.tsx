import { render, screen } from "@testing-library/react";
import React from "react";
import { MockedProvider } from "@apollo/client/testing";
import { defaultMockData } from "../../../../__mocks__/servicesTapMockData";
import ServicesTab from "./ServicesTab";

describe("ServicesTab", () => {
  it("renders loading and loaded state", async () => {
    const { container } = render(
      <MockedProvider mocks={defaultMockData} addTypename={false}>
        <ServicesTab />
      </MockedProvider>,
    );

    expect(await screen.findByTestId("common-spinner")).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
    expect(await screen.findByText("Titre du bloc *")).toBeInTheDocument();
    expect(
      await screen.findByText(
        "Vous pouvez choisir d’afficher jusqu’à 6 blocs maximum",
      ),
    ).toBeInTheDocument();
    expect(
      await screen.findByDisplayValue("Les services de ma ville"),
    ).toBeInTheDocument();
    expect(await container).toMatchSnapshot();
  });
});
