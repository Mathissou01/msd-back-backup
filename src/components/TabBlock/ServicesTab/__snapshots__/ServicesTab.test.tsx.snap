// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServicesTab renders loading and loaded state 1`] = `
<div>
  <div
    aria-labelledby="tab-services"
    class="c-ServicesTab"
    id="panel-services"
    role="tabpanel"
  >
    <div
      class="c-CommonLoader c-CommonLoader_cover c-CommonLoader_grow"
    >
      <div
        class="c-CommonSpinner"
        data-testid="common-spinner"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;

exports[`ServicesTab renders loading and loaded state 2`] = `
<div>
  <div
    aria-labelledby="tab-services"
    class="c-ServicesTab"
    id="panel-services"
    role="tabpanel"
  >
    <h2
      class="c-ServicesTab__Title"
    >
      Services
    </h2>
    <form
      class="c-ServicesTab__Form"
    >
      <div
        class="c-ServicesTab__Block"
      >
        <div
          class="c-ServicesTab__Group"
        >
          <div
            class="c-FormInput"
          >
            <div
              class="c-FormLabel"
            >
              <label
                class="c-FormLabel__Label"
                for="titleContent"
              >
                <span>
                  Titre du bloc *
                </span>
              </label>
              <div
                class="c-FormLabel__Content"
              >
                <input
                  aria-errormessage="titleContent_error"
                  aria-invalid="false"
                  class="c-FormInput__Input"
                  data-testid="form-input"
                  id="titleContent"
                  name="titleContent"
                  type="text"
                  value="Les services de ma ville"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c-ServicesTab__ServiceLinks"
      >
        <div
          class="c-FormServiceLinks"
        >
          <div
            class="c-FormLabel"
          >
            <label
              class="c-FormLabel__Label"
            >
              <span>
                Ordre d’affichage des services sur la page d’accueil (accès rapide)
              </span>
            </label>
            <label
              class="c-FormLabel__Label"
            >
              <span
                class="c-FormLabel__Secondary"
              >
                Vous pouvez choisir d’afficher jusqu’à 6 blocs maximum, à choisir en fonction du menu défini préalablement
              </span>
            </label>
          </div>
        </div>
        <div
          class="c-ServicesTabAddButton"
        >
          <button
            class="c-CommonButton c-CommonButton_secondary c-CommonButton_fontSmall"
            type="button"
          >
            <div
              class="c-CommonButton__Picto c-CommonButton__Picto_cross"
            />
            Ajouter un service personnalisé
          </button>
        </div>
      </div>
      <div
        class="c-ServicesTab__Buttons"
      >
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_primary c-CommonButton_fontSmall"
          disabled=""
          type="submit"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_check"
          />
          Enregistrer les modifications
        </button>
        <button
          class="c-CommonButton c-CommonButton_disabled c-CommonButton_secondary c-CommonButton_fontSmall"
          disabled=""
          type="button"
        >
          <div
            class="c-CommonButton__Picto c-CommonButton__Picto_cross"
          />
          Annuler les modifications
        </button>
      </div>
    </form>
  </div>
</div>
`;
